pkg/asset/installconfig/gcp/client.go

```diff
@@ -80,6 +82,62 @@ func (c *Client) GetMachineType(ctx context.Context, project, zone, machineType
 	return req, nil
 }
 
+// GetMachineTypeList retrieves the machine type with the specified fields.
+func GetMachineTypeList(ctx context.Context, svc *compute.Service, project, region, machineType, fields string) ([]*compute.MachineType, error) {
+	var machines []*compute.MachineType
+
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	filter := fmt.Sprintf("name = \"%s\" AND zone : %s-*", machineType, region)
```

Comment:
text says zone but variable is region? Mismatch or is that how the query works?

Reply:
ah got it. this would work.

Reply:
Yeah, the zone name is prefixed by the region's name.